terraform/variables/README.md

```diff
@@ -0,0 +1,205 @@
+```shell
+$ cd ./terraform/main
+$ terraform init
+...
+$ terraform apply
+...
+Apply complete! Resources: 0 added, 0 changed, 0 destroyed.
+
+Outputs:
+
+list_of_strings = [
+  "item_1",
+  "item_2",
+  "item_3",
+]
+map_of_strings = {
+  "key1" = "value1"
+  "key2" = "value2"
+}
+simple_string = "justastring"
+```
+
+## Waypoint setup
+
+#### Waypoint CLI
+
+Install the Waypoint CLI. See [installation instructions](https://developer.hashicorp.com/waypoint/tutorials/get-started-docker/get-started-install).
+
+#### Waypoint Server
+
+You'll need a Waypoint server. We recommend using [HCP Waypoint](https://cloud.hashicorp.com/products/waypoint)
+You'll also need your CLI configured with a [context](https://developer.hashicorp.com/waypoint/commands/context-create)
+pointing to this server.
+
+#### Waypoint remote runner
+
+Waypoint remote runners are required to use variables with dynamic defaults. HCP Waypoint will 
+walk you through installing a remote runner into your infrastructure. If you don't have
+a runner already, you can run `waypoint runner agent` in a separate terminal window. This
+runner will have access to docker on your laptop, which is where we'll be deploying.
+
+#### Waypoint Terraform Configsourcer Config
+
+Waypoint needs to know how to talk to Terraform cloud at runtime to source the outputs. 
+To accomplish this, run:
+
+```shell
+waypoint config source-set -type=terraform-cloud -config=token=<your-tfc-token>
+```
+#### Customize the waypoint.hcl
+
+
+Take a look at the waypoint.hcl. In each variable stanza, you'll need to update the
+Terraform organization to match the one your chose during the Terraform setup
+
+Example
+```hcl
+variable "single_string_output" {
+  default = dynamic("terraform-cloud", {
+    organization = "waypoint_tfc_vars_example" // REPLACE WITH YOUR ORG ID
+    workspace    = "waypoint_tfc_vars_example"
+    output       = "simple_string"
+  })
+  type        = string
+  sensitive   = false
+  description = "Just one tfc output variable with a string type"
+}
+```
+
+You'll need to commit those changes, and then push to a repository you control. We recommend
+forking the waypoint-examples repo and modifying your fork. The changes need to be 
+pushed for the remote runner to recognize them.
+
+#### Waypoint Project
+
+Create your Waypoint project, pointed to your git repo. You'll need to update the git
+url to point to your fork
+
+```shell
+$ waypoint project apply \
+  -data-source="git" \
+  -git-ref=main \
+  -git-url="https://github.com/<YOUR_ORG>/waypoint-examples" \ # <-- YOUR FORK HERE
+  -git-path=terraform/variables \
+  waypoint_tfc_vars_example
+  
+✓ Project "waypoint_tfc_vars_example" created
+```
+
+## Running the deployment
+
+Run `waypoint up` to build and deploy a docker container. If you're running a runner locally
+using `waypoint runner agent` (and you have no other runners), add `-local=false` to force
+remote execution on that runner. 
+
+```shell
+$ wp up -local=false
```

Comment:
```suggestion
$ waypoint up -local=false
```
It might be worth putting the fully binary name over an alias to keep consistency throughout the README. This way too it is easier on users copying/pasting.

Reply:
thank you!

Reply:
Agreed, full binary name should be used over any aliases.